Sources/HummingbirdWebSocket/Client/WebSocketClient.swift

```diff
@@ -0,0 +1,231 @@
+    public init(
+        url: HBURL,
+        transportServicesTLSOptions: TSTLSOptions,
+        maxFrameSize: Int = 1 << 14,
+        eventLoopGroup: NIOTSEventLoopGroup = NIOTSEventLoopGroup.singleton,
+        logger: Logger,
+        handlerCallback: @escaping HBWebSocketDataCallbackHandler.Callback
+    ) throws where Handler == HBWebSocketDataCallbackHandler {
+        let handler = HBWebSocketDataCallbackHandler(handlerCallback)
+        try self.init(
+            url: url, 
+            transportServicesTLSOptions: transportServicesTLSOptions,
+            handler: handler, 
+            maxFrameSize: maxFrameSize, 
+            eventLoopGroup: eventLoopGroup, 
+            logger: logger
+        )
+    }
+#endif
+
+    ///  Connect and run handler
+    public func run() async throws {
+        guard let host = url.host else { throw Error.invalidURL }
+        let requiresTLS = url.scheme == .wss || url.scheme == .https
+        let port = url.port ?? (requiresTLS ? 443 : 80)
+        if requiresTLS {
+            switch self.tlsConfiguration {
+            case .niossl(let tlsConfiguration):
+                let client = HBClient(
+                    try TLSClientChannel(
+                        WebSocketClientChannel(handler: handler, maxFrameSize: maxFrameSize), 
+                        tlsConfiguration: tlsConfiguration
+                    ),
+                    address: .hostname(host, port: port),
+                    eventLoopGroup: eventLoopGroup,
+                    logger: logger
+                )
+                try await client.run()
+
+            #if canImport(Network)
+            case .ts(let tlsOptions):
+                let client = HBClient(
+                    WebSocketClientChannel(handler: handler, maxFrameSize: maxFrameSize),
+                    address: .hostname(host, port: port),
+                    transportServicesTLSOptions: tlsOptions,
+                    eventLoopGroup: eventLoopGroup,
+                    logger: logger
+                )
+                try await client.run()
+
+            #endif
+            case .none:
+                let client = HBClient(
+                    try TLSClientChannel(
+                        WebSocketClientChannel(handler: handler, maxFrameSize: maxFrameSize), 
+                        tlsConfiguration: TLSConfiguration.makeClientConfiguration()
+                    ),
+                    address: .hostname(host, port: port),
+                    eventLoopGroup: eventLoopGroup,
+                    logger: logger
+                )
+                try await client.run()
+            }
+        } else {
+            let client = HBClient(
+                WebSocketClientChannel(handler: handler, maxFrameSize: maxFrameSize),
+                address: .hostname(host, port: port),
+                eventLoopGroup: eventLoopGroup,
+                logger: logger
+            )
+            try await client.run()
```

Comment:
I don't like the repetitiveness of creating a client and running it. Would it make sense to add a generic `func run(on channel: some HBClientChannel)` that we can call for each of these cases?

Reply:
I wrote it like that to conform it to `Service` but then forgot to add the conformance.
We could add a helper function `HBWebSocket.connect`, which does the creation and run in one go.

Reply:
So I added a `connect` function
```swift
try await HBWebSocketClient.connect(url: .init("ws://test.com/ws"), logger: logger) { inbound, _, _ in
    var inboundIterator = inbound.makeAsyncIterator()
    let msg = await inboundIterator.next()
    XCTAssertEqual(msg, .text("Hello"))
}
```